devito/dimension.py

```diff
@@ -32,6 +32,30 @@ def symbolic_size(self):
         """The symbolic size of this dimension."""
         return self.rtargs[0].as_symbol
 
+    @property
+    def symbolic_extent(self):
+        """Return the extent of the loop over this dimension.
```

Comment:
I wouldn't use the work "loop" here

Reply:
BTW, random thought, I think we should start calling these objects ``Iterators``, or at least something better than ``Dimension``. Or we should derive Iterators from Dimensions... don't know. Just a random thought

Reply:
We already derive `Iteration`s from `Dimension`s. How are `Iterator`s different from `Iteration`s?

Reply:
It's just that at this point I'm not sure I like the name ``Dimension`` given what these objects can do...but this is more than debatable